### ISSUE TITLE
[2020.03] PHP Warnings

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#7999]
Please report here any PHP Warning message obtained running the `2019.12` stable or `2020.03-dev` versions.

Follow-up to #7676

**Comment by MrPetovan** [comment]
[Submitted](https://github.com/friendica/friendica/issues/7676#issuecomment-538666825) by @AlfredSK
> 2019.09 master:
> 
> ```
> PHP Warning:  A non-numeric value encountered in .../mod/profile_photo.php on line 72
> PHP Warning:  A non-numeric value encountered in .../mod/profile_photo.php on line 73
> PHP Warning:  Imagick::cropimage() expects parameter 3 to be integer, string given in .../src/Object/Image.php on line 612
> ```

**Comment by VVelox** [comment]
Howdy! Seeing this on the most recent stable randomly when bin/worker.php is ran by cron.

```
PHP Warning:  Error while sending STMT_CLOSE packet. PID=32080 in /home/goatdaddy/www/src/Database/Database.php on line 666
```

**Comment by MrPetovan** [comment]
This is an indication the database closed the connection early and might be a symptom for resource starvation.

**Comment by tobiasd** [comment]
In addition to https://github.com/friendica/friendica/issues/7998#issuecomment-573290305 there is one other warning in the PHP log file

    PHP Warning:  include(/vendor/composer/../../src/Core/Cache.php): failed to open stream: No such file or directory in /vendor/composer/ClassLoader.php on line 444

**Comment by MrPetovan** [comment]
Likely a race condition between an update of the code and running Composer. The class file would have been removed before the Composer autoload optimizer ran.

**Comment by VVelox** [comment]
> This is an indication the database closed the connection early and might be a symptom for resource starvation.

Well figured out what has been causing that occasionally....

> 2020-01-12T12:32:06.788228Z 4 [Note] Aborted connection 4 to db: 'friendica' user: 'friendica' host: 'vulpes' (Got a packet bigger than 'max_allowed_packet' bytes)

**Comment by AlfredSK** [comment]
The DB server setting 'max_allowed_packet' should be configured with 16M. That should work.

**Comment by VVelox** [comment]
It works nearly consistently except the rare chance it throws that error.

**Comment by VVelox** [comment]
That said, it was at the default. Bumped it up to 128M, which I figure should cover anything.

**Comment by nupplaphil** [comment]
Right after accepting a contact from the nerdica instance:
```
Warning: openssl_public_encrypt(): key parameter is not a valid public key in /var/www/html/mod/dfrn_confirm.php on line 182
```

Additional information from the remote:

> Unexpected response from remote site:
> <!DOCTYPE html > <html> <head> <title>Nerdica | Dfrn_confirm</title> <meta request="dfrn_confirm/lorenzo"> <script type="text/javascript">var baseurl = "https://nerdica.net";</script> <script type="text/javascript">var frio = "view/theme/frio";</script> <meta http-equiv="Content-Type" content="text/html;charset=utf-8" /> <base href="https://nerdica.net/" /> <meta name="generator" content="Friendica 2019.12" /> <meta name="viewport" content="initial-scale=1.0"> <link rel="stylesheet" href="view/global.css" type="text/css" media="all" /> <link rel="stylesheet" href="view/asset/jquery-colorbox/example5/colorbox.css" type="text/css" media="screen" /> <link rel="stylesheet" href="view/asset/jgrowl/jquery.jgrowl.min.css" type="text/css" media="screen" /> <link rel="stylesheet" href="view/asset/jquery-datetimepicker/build/jquery.datetimepicker.min.css" type="text/css" media="screen" /> <link rel="stylesheet" href="view/asset/perfect-scrollbar/css/perfect-scrollbar.min.css" type="text/css" media="screen" /> <link rel="stylesheet" href="view/theme/frio/frameworks/bootstrap/css/bootstrap.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/bootstrap/css/bootstrap-theme.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/asset/fork-awesome/css/fork-awesome.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/jasny/css/jasny-bootstrap.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/bootstrap-select/css/bootstrap-select.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/ekko-lightbox/ekko-lightbox.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/awesome-bootstrap-checkbox/awesome-bootstrap-checkbox.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/justifiedGallery/justifiedGallery.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/bootstrap-colorpicker/css/bootstrap-colorpicker.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/frameworks/bootstrap-toggle/css/bootstrap-toggle.min.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/font/open_sans/open-sans.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/style.pcss" type="text/css" media="screen" /> <link rel="stylesheet" href="view/theme/frio/css/hovercard.css" type="text/css" media="screen"/> <link rel="stylesheet" href="view/theme/frio/css/font-awesome.custom.css" type="text/css" media="screen"/> <link rel="shortcut icon" href="images/friendica-32.png" /> <link rel="apple-touch-icon" href="images/friendica-128.png"/> <meta name="apple-mobile-web-app-capable" content="yes" /> <link rel="manifest" href="https://nerdica.net/manifest" /> <script type="text/javascript"> // Prevents links to switch to Safari in a home screen app - see https://gist.github.com/irae/1042167 (function(a,b,c){if(c in b&&b[c]){var d,e=a.location,f=/^(a|html)$/i;a.addEventListener("click",function(a){d=a.target;while(!f.test(d.nodeName))d=d.parentNode;"href"in d&&(chref=d.href).replace("https://nerdica.net/", "").replace(e.href,"").indexOf("#")&&(!/^[a-z\+\.\-]+:/i.test(chref)||chref.indexOf(e.protocol+"//"+e.host)===0)&&(a.preventDefault(),e.href=d.href)},!1)}})(document,window.navigator,"standalone"); </script> <link rel="search" href="https://nerdica.net/opensearch" type="application/opensearchdescription+xml" title="Search in Friendica" />  <script type="text/javascript" src="view/js/modernizr.js"></script> <script type="text/javascript" src="view/asset/jquery/dist/jquery.min.js"></script> <script type="text/javascript" src="view/js/jquery.textinputs.js"></script> <script type="text/javascript" src="view/js/jquery-textcomplete/jquery.textcomplete.min.js"></script> <script type="text/javascript" src="view/js/autocomplete.js"></script> <script type="text/javascript" src="view/asset/jquery-colorbox/jquery.colorbox-min.js"></script> <script type="text/javascript" src="view/asset/jgrowl/jquery.jgrowl.min.js"></script> <script type="text/javascript" src="view/asset/jquery-datetimepicker/build/jquery.datetimepicker.full.min.js"></script> <script type="text/javascript" src="view/asset/perfect-scrollbar/js/perfect-scrollbar.jquery.min.js"></script> <script type="text/javascript" src="view/asset/imagesloaded/imagesloaded.pkgd.min.js"></script> <script type="text/javascript" src="view/asset/base64/base64.min.js"></script> <script type="text/javascript" src="view/asset/dompurify/dist/purify.min.js"></script> <script type="text/javascript" src="view/js/main.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/bootstrap/js/bootstrap.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/jasny/js/jasny-bootstrap.custom.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/bootstrap-select/js/bootstrap-select.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/ekko-lightbox/ekko-lightbox.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/justifiedGallery/jquery.justifiedGallery.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/bootstrap-colorpicker/js/bootstrap-colorpicker.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/flexMenu/flexmenu.custom.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/jquery-scrollspy/jquery-scrollspy.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/autosize/autosize.min.js"></script> <script type="text/javascript" src="view/theme/frio/frameworks/sticky-kit/jquery.sticky-kit.min.js"></script> <script type="text/javascript" src="view/theme/frio/js/theme.js"></script> <script type="text/javascript" src="view/theme/frio/js/modal.js"></script> <script type="text/javascript" src="view/theme/frio/js/hovercard.js"></script> <script type="text/javascript" src="view/theme/frio/js/textedit.js"></script> <script type="text/javascript"> var updateInterval = 40000; var localUser = false; var aStr = { 'delitem' : "Delete this item?", }; </script> <script type="text/javascript"> var compose = 0; </script><script type="text/javascript"> $(document).ready(function() { $("#nav-search-input-field").search_autocomplete(baseurl + '/search/acl'); }); </script> <script type="text/javascript" src="https://nerdica.net/addon/openstreetmap/openstreetmap.js"></script> <meta name="theme-color" content="#708fa0" /> </head> <body id="top" class="mod-dfrn_confirm is-not-singleuser desktop-view"> <a href="#content" class="sr-only sr-only-focusable">Skip to main content</a> <div id="panel" style="display: none;"></div> <nav class="navbar navbar-fixed-top"> <div class="container"> <div class="navbar-header pull-left"> <a class="navbar-brand" href="#"><div id="navbrand-container"> <div id="logo-img"></div> <div id="navbar-brand-text"> Friendica</div></div> </a> </div> <div class="pull-right"> <ul class="nav navbar-nav navbar-right"> <li role="presentation"> <a href="login?mode=none" id="nav-login" data-toggle="tooltip" aria-label="Sign in" title="Sign in"> <i class="fa fa-sign-in fa-fw" aria-hidden="true"></i> </a> </li> <li role="presentation"> <a href="friendica" id="nav-about" data-toggle="tooltip" aria-label="Information about this Friendica instance" title="Information about this Friendica instance"> <i class="fa fa-info fa-fw" aria-hidden="true"></i> </a> </li> </ul> </div> </div> </nav> <div id="search-mobile" class="hidden-lg hidden-md collapse"> <form class="navbar-form" role="search" method="get" action="search">  <div class="form-group form-group-search"> <input id="nav-search-input-field-mobile" class="form-control form-search" type="text" name="q" data-toggle="tooltip" title="@name, !forum, #tags, content" placeholder="Search"> <button class="btn btn-default btn-sm form-button-search" type="submit">Search</button> </div> </form> </div> <div id="topbar-second" class="topbar"> <div class="container"> <div class="col-lg-3 col-md-3 hidden-sm hidden-xs" id="nav-short-info"></div> <div class="col-lg-7 col-md-7 col-sm-11 col-xs-10" id="tabmenu"></div> <div class="col-lg-2 col-md-2 col-sm-1 col-xs-2" id="navbar-button"></div> </div> </div> <svg id="friendica-logo-mask" x="0px" y="0px" width="0px" height="0px" viewBox="0 0 250 250"> <defs> <mask id="logo-mask" maskUnits="objectBoundingBox" maskContentUnits="objectBoundingBox"> <path style="fill-rule:evenodd;clip-rule:evenodd;fill:#ffffff;" d="M0.796,0L0.172,0.004C0.068,0.008,0.008,0.068,0,0.172V0.824c0,0.076,0.06,0.16,0.168,0.172h0.652c0.072,0,0.148-0.06,0.172-0.144V0.14C1,0.06,0.908,0,0.796,0zM0.812,0.968H0.36v-0.224h0.312v-0.24H0.36V0.3h0.316l0-0.264l0.116-0c0.088,0,0.164,0.044,0.164,0.096l0,0.696C0.96,0.912,0.876,0.968,0.812,0.968z"></path> </mask> </defs> </svg> <main> <div class="container"> <div class="row"> <aside class="col-lg-3 col-md-3 offcanvas-sm offcanvas-xs"> </aside> <div class="col-lg-7 col-md-7 col-sm-12 col-xs-12" id="content"> <section class="sectiontop dfrn_confirm-content-wrapper"> <div id="pause"></div>  </section> </div> </div> </div> <div id="back-to-top" title="back to top">&#8679;</div> </main> <footer>  <div id="modal" class="modal fade" tabindex="-1" role="dialog"> <div class="modal-dialog modal-full-screen"> <div class="modal-content"> <div id="modal-header" class="modal-header"> <button id="modal-cloase" type="button" class="close" data-dismiss="modal"> &times; </button> <h4 id="modal-title" class="modal-title"></h4> </div> <div id="modal-body" class="modal-body">  </div> </div> </div> </div>  <div id="cache-container"></div> <script type="text/javascript" src="addon/mathjax/asset/MathJax.js?config=TeX-MML-AM_CHTML"></script> <script type="text/javascript" src="addon/mathjax/mathjax.js"></script> </footer>  </body>

**Comment by AlfredSK** [comment]
develop:

`PHP Warning:  Illegal string offset 'share' in .../view/smarty3/compiled/4d360747983cf4195b3c0a1f61496a405b0f8b25_2.file.wall_thread.tpl.php on line 643`

I deleted the compiled files to start from scratch but the warnings are still being logged.

**Comment by AlfredSK** [comment]
develop:

`htmlspecialchars() expects parameter 1 to be string, array given in .../view/smarty3/compiled/f6b28614aa6629eb375a0f99a9554f2cc42f1dbf_2.file.field_checkbox.tpl.php on line 33`

**Comment by MrPetovan** [comment]
Still? Grmbl.

**Comment by AlfredSK** [comment]
Haha. Sorry. 
It's difficult to find out. The timezone of the error message is Europe/Berlin. I'm using UTC. So, it wasn't me who triggered it. Maybe I can provoke the error somehow. Let's see...

**Comment by wiwie** [comment]
I have the same error. It's occurring on several pages, example: admin -> additional features -> one warning for each checkbox it seems

**Comment by MrPetovan** [comment]
On the latest develop?

**Comment by wiwie** [comment]
Yes. Pulled yesterday I think, maybe even tonight

Am 29. Februar 2020 10:13:03 schrieb Hypolite Petovan 
<notifications@github.com>:

**Comment by AlfredSK** [comment]
Yup! I can confirm that the warnings appear when I visit `.../admin/features`.

**Comment by MrPetovan** [comment]
Thanks, it's a big help!

**Comment by MrPetovan** [comment]
Closed by #8474
